Review test workflow

KristjanESPERANTO opened the pull request:
- Run linting only once
- Update actions/cache to v4
- Run e2e-tests with <del>the same matrix as the others</del> node 18 instead of node 16
- Change cache keys to avoid cache collisions

traines-source commented:
Thanks, looks reasonable. But I don't think we should run the e2e tests multiple times, since they go against the live DB APIs (and as happening in this run, they are quite unstable, although that's maybe only due to my change of the HAFAS journeys IDs/refreshCtx where the departure times seem to toggle (?)). But probably it makes sense to run them on a newer node version :)

KristjanESPERANTO commented:
> But I don't think we should run the e2e tests multiple times, since they go against the live DB APIs

Okay, that's a good point against my change. I've just changed it back to run only once.

> But probably it makes sense to run them on a newer node version :)

Yeah, I changed it to node 18, which is the oldest node version that hasn't reached End Of Life yet.
